### PR TITLE
[Tyr] Push fusio data to S3/Minio

### PR DESCRIPTION
**Pull request by AzHicham**
NAV-1124

### CODE REVIEW

File: source/tyr/tyr/binarisation.py [hunk 1]
```diff
@@ -1047,3 +1066,69 @@ def poi2mimir(self, instance_name, input, autocomplete_version, job_id=None, dat
...
+def fusio2s3(self, instance_config, filename, job_id, dataset_uid):
+    """ Zip fusio file and launch fusio2s3 """
+
+    job = models.Job.query.get(job_id)
+    dataset = _retrieve_dataset_and_set_state("fusio", job.id)
+    instance = job.instance
+
+    logger = get_instance_logger(instance, task_id=job_id)
+    try:
+        filename = zip_if_needed(filename)
```
Comment (azime) [inline source/tyr/tyr/binarisation.py]: I'm not sure if the zip is necessary

File: source/tyr/tyr/binarisation.py [hunk 2]
```diff
@@ -1047,3 +1066,69 @@ def poi2mimir(self, instance_name, input, autocomplete_version, job_id=None, dat
...
+
+        config = MinioConfig()
+        client = Minio(endpoint=config.host, access_key=config.key, secret_key=config.secret, secure=False)
+
+        dt_now_str = datetime.datetime.now().strftime("%Y-%m-%d-%H:%M:%S")
+        tags = {"coverage": instance_config.name, "datetime": dt_now_str, "data_type": "gtfs"}
```
Comment (azime) [inline source/tyr/tyr/binarisation.py]:
```suggestion
        tags = {"coverage": instance_config.name, "datetime": dt_now_str, "data_type": "ntfs"}
```

File: source/tyr/tyr/binarisation.py [hunk 3]
```diff
@@ -1047,3 +1066,69 @@ def poi2mimir(self, instance_name, input, autocomplete_version, job_id=None, dat
...
+        logger.exception("")
+        job.state = "failed"
+        dataset.state = "failed"
+        raise
+    finally:
+        models.db.session.commit()
+
+
+@celery.task(bind=True)
+def gtfs2s3(self, instance_config, filename, job_id, dataset_uid):
```
Comment (azime) [inline source/tyr/tyr/binarisation.py:1078]: The two functions are similar(gtfs2s3, fusio2s3), you can't do the refacto?

File: source/tyr/requirements.txt [hunk 4]
```diff
@@ -37,3 +37,4 @@ kombu==4.0.2 ; python_version < "3.9"
 redis==2.10.6
 celery==4.1.0 ; python_version < "3.9"
 celery==5.2.2 ; python_version >= "3.9"
+minio==6.0.2
```
Comment (xlqian) [inline source/tyr/requirements.txt:40]: :(

File: source/tyr/tyr/binarisation.py (hunk 1 above)
Comment (AzHicham) [reply source/tyr/tyr/binarisation.py]: If we receive a zip we do nothing, it's just a security in the case we receive a folder of fusio/gtfs data

File: source/tyr/tyr/binarisation.py [hunk 5]
```diff
@@ -1047,3 +1066,49 @@ def poi2mimir(self, instance_name, input, autocomplete_version, job_id=None, dat
...
+    """ Zip fusio file and launch gtfs2s3 """
+    _inner_2s3(self, "gtfs", instance_config, filename, job_id, dataset_uid)
+
+
+def _inner_2s3(self, dataset_type, instance_config, filename, job_id, dataset_uid):
+    """ Zip fusio file and launch gtfs2s3 """
```
Comment (pbench) [inline source/tyr/tyr/binarisation.py]:
```suggestion
```

File: source/tyr/tyr/tasks.py [hunk 6]
```diff
@@ -176,6 +178,14 @@ def process_ed2nav():
                 filename = move_to_backupdirectory(_file, instance_config.backup_directory, manage_sp_char=True)
             else:
                 filename = _file
+
+            is_pt_planner_loki = (
+                hasattr(instance_config, 'pt_planner_id') and instance_config.pt_planner_id == "loki"
```
Comment (pbench) [inline source/tyr/tyr/tasks.py]: look into 'pt_planner_configurations' instead of 'pt_planner_id'

**Review by pbench — APPROVED** [review]

**Review by azime — APPROVED** [review]

**Pull request merged**